src/NexusMods.DataModel/Settings/DataStoreSettingsBackend.cs

```diff
@@ -0,0 +1,84 @@
+using System.Text.Json;
+using JetBrains.Annotations;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using NexusMods.Abstractions.Serialization;
+using NexusMods.Abstractions.Serialization.DataModel;
+using NexusMods.Abstractions.Serialization.DataModel.Ids;
+using NexusMods.Abstractions.Settings;
+
+namespace NexusMods.DataModel.Settings;
+
+[UsedImplicitly]
+internal sealed class DataStoreSettingsBackend : ISettingsStorageBackend
+{
+    private readonly ILogger _logger;
+    private readonly Lazy<IDataStore> _dataStore;
+    private readonly Lazy<JsonSerializerOptions> _jsonOptions;
+
+    public DataStoreSettingsBackend(
+        IServiceProvider serviceProvider,
+        ILogger<DataStoreSettingsBackend> logger)
+    {
+        _logger = logger;
+
+        // NOTE(erri120): Lazy to fix circular dependency:
+        // SettingsManager -> DataStoreSettingsBackend -> IDataStore -> ISettingsManager
+        _dataStore = new Lazy<IDataStore>(serviceProvider.GetRequiredService<IDataStore>);
+        _jsonOptions = new Lazy<JsonSerializerOptions>(serviceProvider.GetRequiredService<JsonSerializerOptions>);
+    }
+
+    public SettingsStorageBackendId Id { get; } = Abstractions.Serialization.Settings.Extensions.StorageBackendId;
```

Comment:
This doesn't make sense to me.
This can't be how we are supposed to define new backends.